index.js

```diff
@@ -281,12 +281,20 @@ function decorateFunctions(node) {
     // Pair up the function params with the JSDoc params (if they exist)
     funcNode.node.params.forEach(function(param) {
         for (i = 0; i < funcNode.jsdoc.params.length; i++) {
-            if (funcNode.jsdoc.params[i].name === param.name &&
-                    funcNode.jsdoc.params[i].type) {
-                // replace the function param name with the type annotated param
-                param.update(
-                    param.source() + ": " + funcNode.jsdoc.params[i].type
-                );
+            if (funcNode.jsdoc.params[i].type) {
+                if (param.type === 'AssignmentPattern') {
+                    if (funcNode.jsdoc.params[i].name === param.left.name) {
```

Comment:
Interesting possibility of getting conflicting type information here, e.g:

```js
/**
 * @param {Number} foo
 */
function (foo = "hello") {

}
```

We can either:
 - Prefer the type given in the default.
 - Prefer the type given in the JSDoc.
 - Do nothing.
 - Complain and die.

It looks like this is opting to always prefer the JSDoc, and this is fine, but you neglect to comment why. Ideally we should be complaining and dying instead, as the code appears contradictory, but since the default is effectively arbitrary code we cannot catch all cases reliably.

Please can you add a comment explaining the decision you've made.